src/sources/kubernetes/file_source_builder.rs

```diff
@@ -0,0 +1,222 @@
+            && self.config.include_pod_uids.is_empty();
+
+        // Default excludes
+        if no_include {
+            // Since there is no user intention in including specific namespace/pod/container,
+            // exclude kuberenetes and vector logs.
+
+            // This is correct, but on best effort basis filtering out of logs from kuberentes system components.
+            // More specificly, it will work for all Kubernetes 1.14 and higher, and for some bellow that.
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"kube-system_*").into());
+
+            // NOTE: for now exclude images with name vector, it's a rough solution, but necessary for now
+            self.file_config
+                .exclude
+                .push((LOG_DIRECTORY.to_owned() + r"*/vector*").into());
+        }
+    }
+
+    /// Checks included uids for validity, and transforms them to numerical format.
+    fn numerical_pod_uids(&self) -> Result<Vec<String>, BuildError> {
+        let mut include_pod_uids = Vec::new();
+        for pod_uid in &self.config.include_pod_uids {
+            let mut pod_uid = pod_uid.clone();
+
+            // Check if pod_uid contains only UID valid characters
+            if !pod_uid.chars().all(|c| {
+                c.is_numeric() || ('A'..='F').contains(&c) || ('a'..='f').contains(&c) || c == '-'
+            }) {
+                error!(message = "Configuration 'include_pod_uids' contains not UID", uid = ?pod_uid);
+                return Err(BuildError::IllegalCharacterInUid {
+                    uid: pod_uid.clone(),
+                });
+            }
+
+            // Remove dashes from uid so to always have numerical format
+            // of uid.
+            pod_uid.retain(|c| c != '-');
+
+            if pod_uid.chars().count() > 32 {
+                return Err(BuildError::UidToLarge {
+                    uid: pod_uid.clone(),
+                });
+            }
+
+            include_pod_uids.push(pod_uid);
+        }
+
+        Ok(include_pod_uids)
+    }
+}
+
+/// Transforms uid to two UID forms defined by https://tools.ietf.org/html/rfc4122.
+///  * 32 character number (numercial format)
+///  * 8_char-4_char-4_char-4_char-12_char
+/// If uid has less than 32 hexadecimal digits, if will be filled up
+/// with "[0-9A-Fa-f]" per digit up to 32 .
+fn to_uid_forms(uid: &str) -> Vec<String> {
+    let mut it = uid.chars();
+    vec![
+        format!(
+            "{}-{}-{}-{}-{}",
+            char_or_hexadecimal(&mut it, 8),
+            char_or_hexadecimal(&mut it, 4),
+            char_or_hexadecimal(&mut it, 4),
+            char_or_hexadecimal(&mut it, 4),
+            char_or_hexadecimal(&mut it, 12)
+        ),
+        char_or_hexadecimal(&mut uid.chars(), 32),
+    ]
+}
+
+/// Joins n next characters from iterator.
+/// If there is not enough of characters, uses "[0-9A-Fa-f]" instead of a character.
```

Comment:
Again, here I'm trying to understand why we need to implement this logic? This seems a bit complicated and there is no reference to why we are including this.

Reply:
So, one reason lies in us accepting partial UIDs, and the second is the issue of there being two forms of pod_uid formats. These two interact in a way that we need to be quite specific on matching pattern to be correct.  

example. 
We have 
`include_pod_uids = ["a"]`
and a pod with path
/atricity_agent-2345_984710f83a8c743ff99231049bcb2301/busybox/0.log
or 
/984710f83a8c743ff99231049bcb2301/busybox/0.log

If we had some simple patterns like
`/*_*_a*/*/*.log` and `/a*/*/*.log `
we will have false positive matches. In this case the second pattern would match on namespace instead of UID so we would have false positive.

The commented function is used to construct patterns from UIDs, like "a", to match on any UID starting with "a". In this case the pattern would look like `a[0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f]` and `a[0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f]-[0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f]-[0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f]-[0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f]-[0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f][0-9A-Fa-f]`

I can explain further if necessary.

Reply:
Added documentation and some simplification should make this more clear.